Source/ForecastDeskbarView.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Benjamin Amos
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include <Bitmap.h>
+#include <StringView.h>
+#include <String.h>
+#include <MessageRunner.h>
+#include <Looper.h>
+#include <Roster.h>
+
+#include "ForecastDeskbarView.h"
+#include "ForecastView.h"
+#include "App.h"
+#include "Util.h"
+
+const uint32 kUpdateForecastMessage = 'Updt';
+const float kToolTipDelay = 1000000; /*1000000ms = 1s*/
+
+ForecastDeskbarView::ForecastDeskbarView(BRect viewSize, ForecastView* forecastView)
+	:	BView(viewSize, "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW | B_NAVIGABLE | B_PULSE_NEEDED)
+{
+	fForecastView = forecastView;
+}
+
+ForecastDeskbarView::ForecastDeskbarView(BMessage* archive)
+	:	BView(BRect(0, 0, 15, 15), "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW)
+{
+	fForecastView = NULL;
```

Comment:
How does this work? fForecastView is left to NULL here but used in other methods of the class.
This constructor seems to not be needed, since you are calling the other one in instantiate_deskbar_item anyways.